game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1661,45 +1667,33 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
         }
       }
     };
-    steps.add(
-        new IExecutable() {
-          private static final long serialVersionUID = -1150863964807721395L;
 
-          @Override
-          public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-            if (!isOver
-                && RetreatChecks.canAttackerRetreatPlanes(attackingUnits, gameData, isAmphibious)
-                && !RetreatChecks.canAttackerRetreatPartialAmphib(
-                    attackingUnits, gameData, isAmphibious)) {
-              attackerRetreatPlanes(bridge);
-            }
-          }
-        });
-    steps.add(
-        new IExecutable() {
-          private static final long serialVersionUID = -1150863964807721395L;
+    steps.add(offensiveGeneralRetreat);
+    new IExecutable() {
+      private static final long serialVersionUID = -1150863964807721395L;
 
-          @Override
-          public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-            if (!isOver
-                && RetreatChecks.canAttackerRetreatPartialAmphib(
-                    attackingUnits, gameData, isAmphibious)) {
-              attackerRetreatNonAmphibUnits(bridge);
-            }
-          }
-        });
-    steps.add(
-        new IExecutable() {
-          private static final long serialVersionUID = 669349383898975048L;
+      @Override
+      public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+        // Intentionally left blank
```

Comment:
Some background on why this is intentional could be useful. Why is this intentionally left blank?

Reply:
For some reason, the retreat steps were broken up into three separate IExecutables even though only one of them would ever run.  I merged all of them into one step and so two of the old IExecutables no longer have anything to do.

Reply:
Can you update the code with a comment or a 'remove on major release' annotation? This would help us to know what to do with this going forward.

Reply:
I added the annotation to all of the `execute` methods of the old `IExecutables`.  I also add a larger comment to indicate why it is blank.